solutions/remotemonitoring/armtemplates/basic-static-map.json

```diff
@@ -407,6 +413,63 @@
                 "description": "Diagnostics service endpoint url"
             },
             "defaultValue": "Undefined"
+        },
+        "telemetryStorageType": {
+            "type": "string",
+            "defaultValue": "tsi",
+            "allowedValues": [
+                "tsi", "cosmosdb"
+            ],
+            "metadata": {
+                "description": "Storage type for telemetry messages, support TSI and CosmosDb"
+            }
+        },
+        "tsiEnvironmentName": {
+            "type": "string",
+            "defaultValue": "[concat('tsi-', take(uniqueString(subscription().subscriptionId, resourceGroup().id, parameters('solutionName')), 5))]",
+            "metadata": {
+                "description": "TSI environment name"
+            }
+        },
+        "tsiEnvironmentSkuName": {
+            "type": "string",
+            "defaultValue": "S1",
+            "allowedValues": [
+                "S1", "S2"
+            ],
+            "metadata": {
+                "description": "The name of the sku. For more information, see https://azure.microsoft.com/pricing/details/time-series-insights/"
+            }
+        },
+        "tsiEnvironmentSkuCapacity": {
+            "type": "int",
+            "defaultValue": 1,
+            "minValue": 1,
+            "maxValue": 10,
+            "metadata": {
+                "description": "The unit capacity of the Sku. For more information, see https://azure.microsoft.com/pricing/details/time-series-insights/"
+            }
+        },
+        "tsiEnvironmentDataRetentionTime": {
+            "type": "string",
+            "defaultValue": "P30D",
```

Comment:
Did we want to retain data for more than 30 days? @aditidugar did we want to do this for 400 days?

Reply:
The value 30 is used by CF.

Reply:
@jillcary and @zhang-hua I think we've landed on 180 days (6 months). Since there's a max storage capacity (30 GB) we have a tradeoff to make. We would hit that limit after 11 months in the demo scenario (2.7 GB/month estimated), but if we want to give room for people customizing/needing more storage then starting with about 1/2 might be a good balance. Thoughts welcome!